### PR TITLE
Set Content-Length and increase BufferSize

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #6045 

cc @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs [hunk 1]
```diff
@@ -57,15 +57,17 @@ internal enum PreconditionState
...
-            }
 
             var request = context.HttpContext.Request;
             var httpRequestHeaders = request.GetTypedHeaders();
             var response = context.HttpContext.Response;
             var httpResponseHeaders = response.GetTypedHeaders();
+            if (fileLength.HasValue)
+            {
+                SetAcceptRangeHeader(context);
+                response.ContentLength = fileLength.Value;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs]: So is the idea here to set the content length assuming that it's not a 'range' request, and then overwrite the content length later if it is? I noticed there's some other code in this class that sets content length.

This deserves a comment if that's the case.

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ public class FileResultExecutorBase
         private const string AcceptRangeHeaderValue = "bytes";
 
         // default buffer size as defined in BufferedStream type
-        protected const int BufferSize = 0x1000;
+        protected const int BufferSize = 64 * 1024;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs]: Comment isn't accurate.

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs (hunk 1 above)
Comment (jbagga) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs]: Yes that is the case. I will add a comment

**Comment by jbagga** [comment]
@rynowak Updated

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs [hunk 3]
```diff
@@ -65,6 +64,7 @@ internal enum PreconditionState
             if (fileLength.HasValue)
             {
                 SetAcceptRangeHeader(context);
+                // Sets the Content-Length header to the file length. This is overwritten for valid range requests with the length of the range.
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs]: > Sets the Content-Length header to the file length.

This is totally obvious, comments should the things you can't understand from a trivial reading of the code 😆 

> This is overwritten for valid range requests with the length of the range.

I would instead say something like: "Right now we're just setting the content length to the whole file, which means we're assuming this isn't a range request. If this is a range request, the code that processes the range header will change the content length to the size of the range."

You want to explain what the non-obvious things and assumptions are 👍

**Pull request merged**